web/src/chartOptions/segmented/getSegmentsInfo.ts

```diff
@@ -0,0 +1,93 @@
+import {FLU_SEGMENT_COLOURS} from "../../util";
+import {ECFeature, MaxSegmentLength, SegmentCoords, WgsCovPlotDB} from "../../db";
+import {isNil, omitBy} from "lodash";
+
+
+export function getCustomXAxisLabel(value: number, segments: string[], segCoords: SegmentCoords): string {
+  for (let segment of segments) {
+    let {start, end} = segCoords[segment];
+    if (value >= start && value <= end) {
+      let pos = value - start + 1
+      return `${segment}:${pos.toLocaleString()}`;
+    }
+  }
+  return ""
+}
+
+export const getMaxSegmentLength = (db: WgsCovPlotDB) => {
+  let out: MaxSegmentLength = {};
+  for (let segment of db.chartOptions.selectedSegments) {
+    let maxLength = 0;
+    for (let sample of db.chartOptions.selectedSamples) {
+      // @ts-ignore
+      if (!isNil(db.segments_ref_seq[sample][segment])) {
+        let length = db.segments_ref_seq[sample][segment].length;
+        if (maxLength <= length) {
+          maxLength = length;
+        }
+      }
+    }
+    out[segment] = maxLength;
+  }
+  /*Sample may miss some segments in analysis, remove those segment
+  Handel the case when user select only samples that miss some segments in analysis
```

Comment:
@nhhaidee This comment isn't very clear. Are you talking about cases where the user selects samples with no assembled segment sequences?

Reply:
Yes!